src/core/transpile.jl

```diff
@@ -2064,3 +2101,68 @@ function transpile(
 
     return circuit
 end
+
+struct DecomposeControlledGatesTranspiler <: Transpiler end
+
+function transpile(
+    ::DecomposeControlledGatesTranspiler,
+    circuit::QuantumCircuit,
+)::QuantumCircuit
+    qubit_count = get_num_qubits(circuit)
+    output = QuantumCircuit(qubit_count = qubit_count, name = get_name(circuit))
+
+    for instr in get_circuit_instructions(circuit)
+        if instr isa Readout
+            push!(output, instr)
+            continue
+        end
+
+        gate = get_gate_symbol(instr)
+        if gate isa Controlled
+            kernel = get_kernel(gate)
+
+            connected_qubits = get_connected_qubits(instr)
+            if length(connected_qubits) != 2
+                throw(
+                    ArgumentError(
+                        "DecomposeControlledGatesTranspiler is only implemented for single-target single-control Controlled Gates",
+                    ),
+                )
+            end
+
+            if typeof(kernel) == SigmaZ
+                throw(ArgumentError("use control_z() instead of Controlled(SigmaZ)"))
+            end
+
+            if typeof(kernel) == SigmaX
+                throw(ArgumentError("use control_x() instead of Controlled(SigmaX)"))
+            end
+
+            control = connected_qubits[1]
+            target = connected_qubits[2]
+
+            op = get_operator(kernel)
+
+            # store existing global phase, which is cancelled by as_universal_gate()
+            a_0 = get_canonical_global_phase(get_matrix(op))
+            universal_equivalent = as_universal_gate(target, op)
+
+            (a_d, A, B, C) = decompose_universal_to_A_B_C_gates(
+                get_gate_symbol(universal_equivalent),
+                target,
+            )
+
+            α = -a_0 + a_d
```

Comment:
Let's use `\alpha` or `a` in both places for consistency.

Reply:
updated